### PR TITLE
Changes to have stack dump for members in snappy and spark cluster.

### PR DESCRIPTION
**Pull request by sonalsagarwal**
## Changes proposed in this pull request
The changes will allow hydra to take stack dumps at the end of each, before stopping spark/snappy cluster. 
For now, the stack dumps will be taken in all the tests, whether passed or failed.

## Patch testing
Run targeted bts on the branch.

## ReleaseNotes.txt changes
NA

## Other PRs 
NA

### CODE REVIEW

File: dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java [hunk 1]
```diff
@@ -2511,6 +2557,32 @@ public static synchronized void HydraTask_createAndStartSnappyLocator() {
...
+  }
+
+  public void dumpStacks() {
+    Set pids = getPidList();
+    Iterator itr = pids.iterator();
+    while(itr.hasNext()) {
+      String val = (String)itr.next();
+      int pid = Integer.parseInt(val);
+      String host = getPidHost(val);
+      Log.getLogWriter().info("[Sonal] PID is:" + pid + ";host is:" + host);
```
Comment (spillai14) [inline dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java]: please remove personal loggings.

File: dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java [hunk 2]
```diff
@@ -2511,6 +2557,32 @@ public static synchronized void HydraTask_createAndStartSnappyLocator() {
...
+  public static synchronized void HydraTask_dumpStacks() {
+    initSnappyArtifacts();
+    int dumpItr = SnappyPrms.getNumOfStackDumpItrs();
+    for (int i = 0; i < dumpItr; i++) {
+      snappyTest.dumpStacks();
+      if (i < (dumpItr-1)) {
+        try {
+          Thread.sleep(SnappyPrms.getSleepBtwnStackDumps() * 1000);
+        } catch (InterruptedException ie) {
+        }
```
Comment (spillai14) [inline dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java]: The catch block is empty.

File: dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java [hunk 3]
```diff
@@ -1715,6 +1715,52 @@ public void executeProcess(ProcessBuilder pb, File logFile) {
...
+    } catch (IOException e) {
+      String s = "Problem while starting the process : " + pr;
+      throw new TestException(s, e);
+    } catch (InterruptedException e) {
+      String s = "Exception occurred while waiting for the process execution : " + pr;
+      throw new TestException(s, e);
+    }
+  }
+
   protected synchronized void recordSnappyProcessIDinNukeRun(String pName) {
```
Comment (swatisawant) [inline dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java]: I think, this method cab be removed now.

**Comment by sonalsagarwal** [comment]
@kneeraj For now, it is difficult to find the status of the test, passed, failed or hung. SO as a quick fix, it will take dumps in all the tests. The changes for test status is in another PR.

File: dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java (hunk 3 above)
Comment (sonalsagarwal) [reply dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java]: This is still being used by the previous methods, so will keep it.

**Comment by sonalsagarwal** [comment]
Changes were reverted due to HostAgent not started issues. Need to look into it, hence restored the branch.

**Pull request merged**